pgx/src/memcxt.rs

```diff
@@ -181,12 +181,60 @@ pub struct OwnedMemoryContext {
 
 impl Drop for OwnedMemoryContext {
     fn drop(&mut self) {
+        extern "C" {
+            pub fn pgx_MemoryContextIsValid(context: pg_sys::MemoryContext) -> bool;
+        }
+
+        unsafe fn switch_context(
+            cxt: pg_sys::MemoryContext,
+            previous: pg_sys::MemoryContext,
+        ) -> bool {
+            if ptr::eq(pg_sys::CurrentMemoryContext, cxt) {
+                // If the previous context is no longer valid, switch to TopMemoryContext
+                pg_sys::CurrentMemoryContext = if pgx_MemoryContextIsValid(previous) {
```

Comment:
We want to try to reduce the involvement of the C shim in PGX so we can make it easier to build for various environments, and hopefully even enable cross-compiling in some limited scenarios. This embeds it in every context switch. I am not sure if we can accept this.

Reply:
This won't be a concern if https://github.com/tcdi/pgx/pull/947 is merged.

Reply:
Removed the shim use as #947 was merged.